packages/itwinui-css/src/expandable-block/expandable-block.scss

```diff
@@ -195,6 +195,21 @@
   }
 
   .iui-expandable-content {
+    display: grid;
+  }
+
+  &[data-iui-expanded='false'] .iui-expandable-content {
+    grid-template-rows: 0fr;
+    overflow: hidden;
+    transition: grid-template-rows 0.2s, visibility 0.2s;
+    min-height: 0;
+    visibility: hidden;
+    @include iui-expandable-block-content;
+  }
+
+  &[data-iui-expanded='true'] .iui-expandable-content {
+    grid-template-rows: 1fr;
+    visibility: visible;
     @include iui-expandable-block-content;
   }
```

Comment:
I've found a way to keep padding without needing two wrapping divs - it will require the use of `transform: translateY(-100%)` combined with `z-index` to hide the leftover padding under the header only when it's closed.

Also, because this requires changes in a few different elements, it would be best to clean up the CSS to remove the use of all mixins, and use classes directly. The mixins are confusing and unnecessary.

Reply:
like..this? it works but not smooth. also removed mixin

Reply:
it's not smooth because:
(1) you have removed padding from the closed state.
(2) you are not transitioning `transform`.

this should work:
```scss
    display: grid;
    grid-template-rows: 0fr;
    overflow: hidden;
    z-index: -1;
    transition: grid-template-rows 0.2s, visibility 0.2s, transform 0.2s;

    > div {
      min-height: 0;
      visibility: hidden;
      transition: visibility 0.2s;
      padding: var(--_iui-expandable-block-content-padding);
    }
  }

  &[data-iui-expanded='true'] .iui-expandable-content {
    grid-template-rows: 1fr;

    > div {
      visibility: visible;
    }
  }

  &:not([data-iui-expanded='true']) .iui-expandable-content {
    transform: translateY(-100%);
  }
```

Reply:
as for the mixins, i really meant removing *all* mixins, not just `iui-expandable-content`.

Reply:
Ok now the content is not visible altogether :-s I tried tweaking around too, setting visible variables and transformY on various divs/classes

Reply:
can you try to paste the exact code i wrote above? it works fine for me.

Reply:
Yea using exact code caused the other issue of content now showing when expanded for me which is why I tweaked around (it is the one being pushed just now, you can check [here](https://itwin.github.io/iTwinUI/1396/react/?path=/story/core-expandableblock--basic)
![expandable block](https://github.com/iTwin/iTwinUI/assets/22462084/30cd3e0c-eb1e-449c-bc1a-7dd6ecdc3a4f)

I think is because z-index is still -1 from default. making it not -1 when `data-iui-expanded='true'` would show the content. But like I mentioned, animation is still not smooth cuz collapsing is instant:
![expandable block z](https://github.com/iTwin/iTwinUI/assets/22462084/bf8670da-807e-4565-8325-ccdab6cfbe8c)

Reply:
try adding `isolation: isolate` to `iui-expandable-block` (root)

Reply:
sorcery?? thanks! that does it

Reply:
the problem was that `z-index: -1` places it underneath the background, so `isolation: isolate` will create a new stacking context which tells it to only place it underneath the elements within this context.

Reply:
ohhh that makes sense!